core/pages/MyAccount.js

```diff
@@ -30,9 +33,10 @@ export default {
       this.$router.push(localizedRoute('/', currentStoreView().storeCode))
     }
   },
-  destroyed () {
+  beforeDestroy () {
     this.$bus.$off('myAccount-before-updateUser', this.onBeforeUpdateUser)
     this.$bus.$off('myAccount-before-changePassword', this.onBeforeChangePassword)
+    this.$bus.$off('user-after-logout')
```

Comment:
If we won't specify `callback` (second argument) - it will remove each listener on `user-after-logout` in EventBus - which would result in really hard to debug side-effect. I think we should put:
```js
() => {
      this.$router.push(localizedRoute('/', currentStoreView().storeCode))
    }
```
In methods, and use pointer to it as the `callback` - same as it is done for `myAccount-before-updateUser` and `myAccount-before-changePassword`.

Doc about it: https://vuejs.org/v2/api/#vm-off-event-callback

Reply:
good point! :) thanks